pkg/blockbuilder/tsdb.go

```diff
@@ -248,17 +251,20 @@ func (b *tsdbBuilder) newTSDB(userID string) (*userTSDB, error) {
 		return nil, err
 	}
 
+	db.DisableCompactions()
+
 	udb.db = db
 
 	return udb, nil
 }
 
-// compactBlocks compacts the blocks of all the TSDBs.
-// It moves all the blocks produced into the shipperDir.
-func (b *tsdbBuilder) compactAndClose(ctx context.Context, shipperDir string) error {
+// compactAndUpload compacts the blocks of all the TSDBs
+// and uploads them.
+func (b *tsdbBuilder) compactAndUpload(ctx context.Context, blockUploaderForUser func(context.Context, string) blockUploader) error {
```

Comment:
Only reason to take a `blockUploaderForUser` here instead of simply uploading is to make writing tests easier with a mock uploader. (might need more changes in BlockBuilder for this, but we can do that when writing tests for BlockBuilder)